From fa8cdb3a32c377b6290d0a92d2522186bcd48293 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@theqtcompany.com>
Date: Tue, 9 Aug 2016 16:21:29 +0200
Subject: Do not depend on Linux 4.5

Avoid using MADV_FREE that was only recently added to Linux. It will fail when
run on older Linux kernels.

Change-Id: I9b0369fb31402f088b2327c12f70dd39f5e4c8c0
Reviewed-by: Peter Varga <pvarga@inf.u-szeged.hu>
---
 chromium/third_party/WebKit/Source/wtf/allocator/PageAllocator.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/3rdparty/chromium/third_party/WebKit/Source/wtf/PageAllocator.cpp b/src/3rdparty/chromium/third_party/WebKit/Source/wtf/PageAllocator.cpp
index 12c9a7b..1639013 100644
--- a/src/3rdparty/chromium/third_party/WebKit/Source/wtf/PageAllocator.cpp
+++ b/src/3rdparty/chromium/third_party/WebKit/Source/wtf/PageAllocator.cpp
@@ -39,6 +39,11 @@

 #include <sys/mman.h>
 
+#if OS(LINUX) && defined(MADV_FREE)
+// Added in Linux 4.5, but we don't want to depend on 4.5 at runtime
+#undef MADV_FREE
+#endif
+
 #ifndef MADV_FREE
 #define MADV_FREE MADV_DONTNEED
 #endif
-- 
cgit v1.0-4-g1e03

